### DISCUSSION TITLE
播放器的播放/暂停图标未按视频播放状态更新

### DISCUSSION THREAD

**Discussion by korxy** [hooke007/mpv_PlayKit#322] [mpv 通用]
任意打开一个视频文件，此时视频已经开始播放，但mpv-lazy主程序的播放/暂停显示的还是暂停状态下的三角图标，如下图所示：
![image](https://github.com/hooke007/MPV_lazy/assets/62867138/cbcec0a4-c44c-4cfb-a693-690dcda8555d)

**Comment by hooke007** [comment]
edit : 新结论（仅处理意见） https://github.com/hooke007/MPV_lazy/discussions/322#discussioncomment-6254242

旧结论：
原本的设计逻辑：仅表述属性`pause`的状态而非“播放状态”（它的状态应该由多个属性共同决定）
这是表示“正在播放”的图标 ▶
这是表示“正在暂停”的图标 ⏸

**Reply by hooke007** [reply]
这是状态指示器，表示属性pause的状态。和你怎么开程序有什么关系？

**Reply by korxy** [reply]
一般的播放器播放视频的时候这里是显示两条竖线的图标，而暂停的时候这里是显示三角图标的。可能这里的设计和我理解的不一样吧。

**Reply by hooke007** [reply]
去翻uosc的对应文档，我默认给的是 play_pause 你要的东西叫 pause_play

**Reply by korxy** [reply]
好的，多谢大佬指导

**Reply by korxy** [reply]
通过修改main.lua中的play_pause 为pause_play即可改回一般播放器的播放/暂停图标显示状态：
![image](https://github.com/hooke007/MPV_lazy/assets/62867138/2319ff0b-592b-4940-bcbb-b3388293fdaf)

**Reply by hooke007** [reply]
作为用户，永远不要触碰脚本本身，学会使用配置文件

**Reply by korxy** [reply]
好的，修改script-opts/uosc.conf配置中的controls里的play_pause 为pause_play也能起到作用

**Reply by korxy** [reply]
后面发现问题还是没有解决，按上面那样子做只不过是把图标的显示反转摆了，事实上，图标还是不会按播放状态发生变化...

**Reply by hooke007** [reply]
> 图标还是不会按播放状态发生变化...

如何复现它，我这里一切正常。
我认为你说的“按播放状态发生变化”和我说的“表示属性pause的状态”不是一回事。
不管怎样，它是属性pause的指示器，不是属性idle的指示器（我认为你说的“播放状态”包含了idle，而mpv在idle状态下存在pause yes no 的两种状态，这意味着加载新文件时是保持暂停还是保持播放）

如果我没理解错你的需求，你想要的是 满足 `idle-active` 或 `eof-reached` 或 `pause` 时显示▶，否则显示⏸。
很显然和我设计这个按钮的逻辑有本质上的不同。controls也不支持上述逻辑的自定义设计

**Reply by korxy** [reply]
非常感谢您的回复！

首先，在修改uosc.conf文件中的值为pause_play后，我尝试直接打开播放器发现播放按钮的图标是pause状态，后面我才想到可能是播放器的默认状态是播放导致的。果不其然，在查看配置后，发现一行“#pause = yes”被注释掉了，也就是说默认是播放状态，而后我将其改为yes后，再次直接打开播放器图标才变回了play状态，所以是这个默认状态才引发了这个问题，并不是没有按播放状态更新图标。

然而，修改这个值后面也带来了一些其他的问题，比如直接打开视频时它默认就变成暂停了，需要手动操作才能播放，这...

而且每次视频播放完后，在点击播放按钮就没反应了...我以为会自动开始重新播放的0..0

最后，我重写了那个lua脚本并自定义了播放按钮加了些自己需要的功能，貌似现在可以达到我的需求了...

**Reply by hooke007** [reply]
> 修改这个值后面也带来了一些其他的问题，比如直接打开视频时它默认就变成暂停了

这不是问题，因为这个选项就是这么用的 https://mpv.io/manual/master/#options-pause

**Reply by korxy** [reply]
确实，但是修改这个值给我带来了其他的问题，但是不改这个值我的需求又得不到解决...

**Reply by hooke007** [reply]
https://github.com/hooke007/MPV_lazy/discussions/322#discussioncomment-6243570
我已经给你提示了，实际上的问题是你不了解mpv的运行机制。
只需要 idle-active 时显示你要的图标即可，完全不需要修改option pause

**Reply by korxy** [reply]
我确实不是很了解...
但我的需求是根据”pause“的状态显示需要的图标，而不是”idle“的状态，因此，如果pause的初始状态不是暂停的话就不能达到我的需求，所以这里必须要修改它。

**Reply by hooke007** [reply]
你的话前后矛盾，你要的难道不是启动时显示 ▶ ？而启动时并不是你认知上的 “pause” 的状态

“视频播放状态”从来不是只根据属性`pause`而决定的；
我也指出了为什么不需要修改option pause，说的更直白点修改它是错误的做法，因此才会产生你认知上的“问题”，而它本不是也不会产生“问题”。

这里已经给出了如何做的方向（这也是常规播放器的逻辑）
> 满足 idle-active 或 eof-reached 或 pause 时显示▶，否则显示⏸。

如果你想明白了上述的意思，“问题”自然就不存在了，但是抱歉我没有这个需求去实现常规播放器的状态。

**Reply by korxy** [reply]
这是我通过参考PotPlayer的播放按钮处理过的mpv：https://wwi.lanzoup.com/i78by0zsaloj

原版是这个：mpv-lazy-20230513.exe （可以在您的发布页找到）

您可以下载原版和这个进行对比，通过观察播放按钮的变化可以看出两者的不同，虽然不一定是最优的处理方式，但它确实是满足了我的需求，能跟常规播放器一样的状态变化。

**Reply by hooke007** [reply]
你的处理方式的问题不再赘述。我也知道怎么正确处理但是我很懒，理由也不再赘述。
我不能也不是在强迫你接受我的观点，只是在做最终结论，用于以后可能出现的“重复问题的引用”
